kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/richmessaging/helpers/KmFormStateHelper.java

```diff
@@ -110,7 +112,20 @@ public static void clearInstance() {
                 for (int i = 0; i < formStateModel.getDateFieldArray().size(); i++) {
                     int key = formStateModel.getDateFieldArray().keyAt(i);
                     KmFormPayloadModel.DateTimePicker dateTimePicker = formPayloadModelList.get(key).getDatePickerModel();
-                    formDataMap.put(dateTimePicker.getLabel(), formStateModel.getDateFieldArray().get(key).toString());  //Might need to convert to formatted date
+
+                    if (KmFormPayloadModel.Type.DATE.getValue().equals(formPayloadModelList.get(key).getType())) {
+                        formDataMap.put(dateTimePicker.getLabel(), KmDateUtils.getFormSerialisedDateFormat(formStateModel.getDateFieldArray().get(key)));
+                    } else if (KmFormPayloadModel.Type.TIME.getValue().equals(formPayloadModelList.get(key).getType())) {
+                        formDataMap.put(dateTimePicker.getLabel(), KmDateUtils.getFormSerialisedTimeFormat(formStateModel.getDateFieldArray().get(key)));
+                    } else {
+                        formDataMap.put(dateTimePicker.getLabel(), KmDateUtils.getFormSerialisedDateTimeFormat(formStateModel.getDateFieldArray().get(key)));
+                    }
+                }
+            }
+
+            if (formStateModel.getDropdownFieldArray() != null) {
+                for (int i = 0; i < formStateModel.getDropdownFieldArray().size(); i++) {
```

Comment:
use for each loop? i think null check might not required in that case

Reply:
Using SparseArray Here. For Each is not applicable in this case.